doc/releases/index.rst

```diff
@@ -1,14 +1,52 @@
 .. _zephyr_release_notes:
 
-Release Notes
-#############
+Releases
+########
 
 Zephyr project is provided as source code and build scripts for different
 target architectures and configurations, and not as a binary image. Updated
 versions of the Zephyr project are released approximately every three-months.
 
 All Zephyr project source code is maintained in a `GitHub repository`_.
 
+Release Life Cycle and Maintenance
+**********************************
+
+Periodic Releases
+=================
+
+The Zephyr project provides periodic releases every 4 months leading to the
+long term support releases approx. every 2 years. Priodic and non-LTS releases
+are maintained with updates, bug fixes and security related updates for at
+least two cycles, meaning that the project supports the most recent two
+releases in addition to the most recent LTS.
+
+Long Term Support and Maintenance
+=================================
+
+A Zephyr :ref:`Long Term Support (LTS) <release_process_lts>` release is
+published every 2 years and is branched and maintained independently from the
+main tree for at least 2.5 years after it was released.
+
+Support and maintenance for an LTS release stops at least half a year
+after the following LTS release is published.
```

Comment:
Re: v1.14 EOL notice, I wonder where we can put a list of currently supported LTS and old LTS and status.

Something like this:

![image](https://user-images.githubusercontent.com/62298616/221931643-7adf026e-ea5d-4a8f-9ecf-ac4d679ee757.png)

(snapshot from https://ubuntu.com/about/release-cycle)

Reply:
@kartben  will look into adding this in a followup PR